### PR TITLE
.Net: Updated planner notebook to use function calling

### PR DESCRIPTION
**Pull request by dmytrostruk**
### Motivation and Context

Resolves: https://github.com/microsoft/semantic-kernel/issues/9306

Since function calling is the new recommended way for planning capabilities, this PR contains a change to update the planner notebook to use function calling.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/notebooks/05-using-function-calling.ipynb
Comment (SergeyMenshykh) [inline dotnet/notebooks/05-using-function-calling.ipynb:1]: Is the 'Auto()' method invocation missing in step 3, `FunctionChoiceBehavior = FunctionChoiceBehavior`, or is the FunctionChoiceBehavior variable assigned somewhere above?

**Review by SergeyMenshykh — APPROVED** [review]

File: dotnet/README.md [hunk 1]
```diff
@@ -77,7 +77,7 @@ requirements and setup instructions.
 3. [Running AI prompts from file](./notebooks/02-running-prompts-from-file.ipynb)
 4. [Creating Semantic Functions at runtime (i.e. inline functions)](./notebooks/03-semantic-function-inline.ipynb)
 5. [Using Kernel Arguments to Build a Chat Experience](./notebooks/04-kernel-arguments-chat.ipynb)
-6. [Creating and Executing Plans](./notebooks/05-using-the-planner.ipynb)
+6. [Introduction to the Planning/Function Calling](./notebooks/05-using-function-calling.ipynb)
```
Comment (markwallace-microsoft) [inline dotnet/README.md:80]: Just `Introduction to Function Calling`, I don't think we need to keep the reference to planning

File: dotnet/notebooks/05-using-function-calling.ipynb [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Introduction to the Planning/Function Calling\n",
+    "\n",
+    "Once you have multiple plugins, you then need a way for your AI agent to use them together to solve a user’s need. This is where planning comes in.\n",
+    "\n",
+    "Early on, Semantic Kernel introduced the concept of planners that used prompts to request the AI to choose which functions to invoke. Since Semantic Kernel was introduced, however, OpenAI introduced a native way for the model to invoke or “call” a function: function calling. Other AI models like Gemini, Claude, and Mistral have since adopted function calling as a core capability, making it a cross-model supported feature.\n",
+    "\n",
+    "Because of these advancements, Semantic Kernel has evolved to use function calling as the primary way to plan and execute tasks.\n",
```
Comment (markwallace-microsoft) [inline dotnet/notebooks/05-using-function-calling.ipynb:14]: I don't think we need to explain the history as that information isn't relevant for someone who is new. 

The Blog post is there for people who are migrating.

File: dotnet/notebooks/05-using-function-calling.ipynb [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+    "\n",
+    "Read more about it [here](https://aka.ms/sk/concepts/planner)."
```
Comment (markwallace-microsoft) [inline dotnet/notebooks/05-using-function-calling.ipynb:16]: Suggest we remove this

File: dotnet/notebooks/05-using-function-calling.ipynb
Comment (dmytrostruk) [reply dotnet/notebooks/05-using-function-calling.ipynb:1]: Good catch, fixed.

**Review by markwallace-microsoft — APPROVED** [review]

**Pull request merged**